internal/rbd/types/volume.go

```diff
@@ -40,4 +42,21 @@ type Volume interface {
 
 	// RemoveFromGroup removes the Volume from the VolumeGroup.
 	RemoveFromGroup(ctx context.Context, vg VolumeGroup) error
+
+	// GetPoolName returns the name of the pool where the volume is stored.
+	GetPoolName() string
```

Comment:
Maybe `Volume` should inherit the `journalledInterface` too?

https://github.com/ceph/ceph-csi/blob/aa88b4c4a01554db6456b971fe387dbd391d0a49/internal/rbd/types/group.go#L26-L38

Reply:
This needs a little more refactoring as journalledObject is specific to volumeGroup now, will address this in follow-up when implementing for group support